rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,116 @@
+    def name(cls) -> Text:
+        """Name of your custom channel."""
+        return "twilio_voice"
+
+    def blueprint(
+        self, on_new_message: Callable[[UserMessage], Awaitable[None]]
+    ) -> Blueprint:
+
+        twilio_voice_webhook = Blueprint(
+            "custom_webhook_{}".format(type(self).__name__),
+            inspect.getmodule(self).__name__,
+        )
+
+        @twilio_voice_webhook.route("/", methods=["GET"])
+        async def health(request: Request) -> HTTPResponse:
+            return response.json({"status": "ok"})
+
+        @twilio_voice_webhook.route("/webhook", methods=["POST"])
+        async def receive(request: Request):
+            sender_id = request.form.get("From")  # method to get sender_id
+            text = request.form.get("SpeechResult")  # method to fetch text
+            input_channel = self.name()  # method to fetch input channel
+            call_status = request.form.get("CallStatus")  # method to fetch call status
+            call_sid = request.form.get("CallSid")  # Get the call identifier
+
+            collector = TwilioVoiceCollectingOutputChannel()
+
+            # Provide an initial greeting to answer the user's call.
+            if (text is None) & (call_status == "ringing"):
+                text = "hello"
+
+            # determine the response.
+            if text is not None:
+                await on_new_message(
+                    UserMessage(
+                        text, collector, sender_id, input_channel=input_channel,
+                    )
+                )
+
+                # Parse the text responses and build the Twilio VoiceResponse.
+                respond_segments = []
+                for message in collector.messages:
+                    respond_segments.append(message["text"])
+                    if "buttons" in message:
+                        for button in message["buttons"]:
+                            respond_segments.append(button["title"])
+                twilio_response = build_twilio_voice_response(respond_segments)
+                return response.text(str(twilio_response), content_type="text/xml")
+            # If the user doesn't respond to the previous message resend the last message.
+            elif text is None:
+                # Get last user utterance from tracker.
+                tracker = request.app.agent.tracker_store.retrieve(sender_id)
+                last_response = next(
+                    (e for e in reversed(tracker.events) if isinstance(e, BotUttered)),
+                    None,
+                )
+
+                # If no previous utterance found say something generic.
+                if last_response is None:
+                    last_response = "I didn't get that."
+                else:
+                    last_response = last_response.text
+
+                twilio_response = build_twilio_voice_response([last_response])
+                return response.text(str(twilio_response), content_type="text/xml")
```

Comment:
how about moving out this block instead of having it duplicate in both `if` and `elif` blocks?